### PR TITLE
WIP: console updates

### PR DESCRIPTION
**Pull request by afshin**
Move `banner` to the `ConsoleModel` and create a getter for `prompt`.

### CODE REVIEW

File: src/console/model.ts [hunk 1]
```diff
@@ -179,12 +179,17 @@ class ConsoleModel implements IConsoleModel {
    * Construct a new console model.
    */
   constructor() {
-    let banner = this.createRawCell();
-    let prompt = this.createCodeCell();
-    banner.input.textEditor.readOnly = true;
+    this._banner = this.createRawCell() as RawCellModel;
```
Comment (blink1073) [inline src/console/model.ts]: Are these typecasts to RawCellModel necessary?

File: src/console/model.ts [hunk 2]
```diff
@@ -226,14 +231,17 @@ class ConsoleModel implements IConsoleModel {
...
   set banner(newValue: string) {
-    if (newValue === this._banner) {
+    if (newValue === this._bannerText) {
       return;
     }
     let oldValue = this._banner;
-    this._banner = newValue;
+    this._bannerText = newValue;
+    if (this._banner !== null) {
+      (this._banner as RawCellModel).input.textEditor.text = newValue;
```
Comment (blink1073) [inline src/console/model.ts]: Other typecast here.

File: src/console/widget.ts [hunk 3]
```diff
@@ -114,6 +114,15 @@ class ConsoleWidget extends Widget {
     return widget;
   }
 
+  /*
+   * The last cell in a console is always a `CodeCellWidget` prompt.
```
Comment (blink1073) [inline src/console/widget.ts:118]: Is there no way to change a cell type in QtConsole?

File: src/console/widget.ts [hunk 4]
```diff
@@ -194,39 +204,21 @@ class ConsoleWidget extends Widget {
    * Handle changes to the notebook model.
    */
   protected onModelChanged(model: IConsoleModel, args: IChangedArgs<any>): void {
```
Comment (blink1073) [inline src/console/widget.ts]: I'd get rid of this altogether until explicitly required.

File: src/console/widget.ts (hunk 3 above)
Comment (afshin) [inline src/console/widget.ts:118]: Not that I can see. I think that it makes sense to define the `prompt` as:
1. the final cell
2. a `CodeCellWidget`

**Comment by blink1073** [comment]
Nice!

**Comment by afshin** [comment]
cf. https://github.com/jupyter/jupyter-js-notebook/issues/170

**Pull request merged**